tiledb/sm/array_schema/test/unit_attribute.cc

```diff
@@ -0,0 +1,202 @@
+/**
+ * @file tiledb/sm/array_schema/test/unit_arribute.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file contains unit tests for the attribute class
+ */
+
+#include <tdb_catch.h>
+
+#include "tiledb/sm/array_schema/attribute.h"
+
+using namespace tiledb::common;
+using namespace tiledb::sm;
+
+/**
+ * Datatype::ANY attributes are always var-sized.
+ */
+TEST_CASE(
```

Comment:
This is four test cases combined into one. There's not any common setup code. The comments explaining the test should be in a `TEST_CASE` description, where they'll be printed out in the test run, accelerating getting the relevant information to the programmer.